src/dune_pkg/lock_dir.ml

```diff
@@ -99,17 +142,67 @@ module Pkg = struct
            { Pkg_info.name; version; dev; source }
          in
          { build_command; deps; install_command; info; exported_env; lock_dir }
+
+  let encode
+      { build_command
+      ; install_command
+      ; deps
+      ; info = { Pkg_info.name = _; version; dev; source }
+      ; lock_dir = _
+      ; exported_env
+      } =
+    let open Dune_lang.Encoder in
+    record_fields
+      [ field Fields.version string version
+      ; field_o Fields.install Dune_lang.Action.encode install_command
+      ; field_o Fields.build Dune_lang.Action.encode build_command
+      ; field_l Fields.deps Package_name.encode deps
+      ; field_o Fields.source Source.encode source
+      ; field Fields.dev bool dev
```

Comment:
Just a stylistic choice, but I would omit fields when they're equal to their default value. All the defaults are controlled by the language version, so this is safe to do.

Reply:
This was an actual bug. `decode` decodes this field as a field with no argument (ie. `(dev)`) whereas `encode` was writing a field with a name and boolean value (`(dev true)`). Switching to using the `field_b` function to match the decoder. I still need to add tests that the encoder and decoder round trip.

Reply:
Hmm, well I think that `(dev true)` or `(dev false)` should still be allowed. I'll take a look